CMakeLists.txt

```diff
@@ -396,6 +396,7 @@ SET(LIB_SOURCE_FILES
     lib/common/memory.c
     lib/common/multithread.c
     lib/common/redis.c
+    lib/common/reverse_tunnel.c
```

Comment:
The implementation depends on h2o.h, so it cannot go into lib/common. lib/core will be a better location.

Though, it might just be sufficient to add code to lib/core/util.c, considering that other logic that setup up server-side HTTP (proxy protocol handler and TLS handshake) exists there.

Reply:
Done in https://github.com/h2o/h2o/pull/3405/commits/b75b2029350ff7af85286e6b4dd729d74baf1930